File: preferencedemo/src/main/java/com/example/preference_demo/MainActivity.java
package com.example.preference_demo;

import android.os.Bundle;
import android.util.Log;
import androidx.appcompat.app.AppCompatActivity;
import com.oplus.flashbackmsgsdk.ButtonMessage;
import com.oplus.flashbackmsgsdk.MessageStub;
import com.oplus.flashbackmsgsdk.MessageTag;
import com.oplus.flashbackmsgsdk.MonitorClient;
import com.oplus.flashbackmsgsdk.MonitorListener;
import com.oplus.flashbackmsgsdk.SendCallback;
import com.oplus.flashbackmsgsdk.ServiceLifecycleCallback;

public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getSupportFragmentManager().beginTransaction().replace(R.id.settings_container, new SettingsFragment()).commit();
        updateMsg();
    }

    MonitorClient client;
    MonitorListener listener;
    MessageStub mStub;

    public void updateMsg() {
        client = MonitorClient.getInstance();
        listener = new MonitorListener() {
            @Override
            public void onChanged(MessageStub stub, int errorCode) {
                mStub = stub;
                Log.e("wenming", "闪回消息回调：" + stub);
            }
        };
        // 查询是否支持闪回2.0
        boolean isSupported = client.isSupported(getApplicationContext());
        client.registerListener(getApplicationContext(), listener);
        client.addServiceStatusCallback(new ServiceLifecycleCallback() {
            @Override
            public void onCreated() {
                Log.e("wenming", "onCreated：");
            }

            @Override
            public void onDestroyed() {
                Log.e("wenming", "onDestroyed：");
            }
        });

        Log.e("wenming", "闪回是否支持：" + isSupported);

        client.send(getApplicationContext(), mStub, ButtonMessage.KEY_BUTTON_CENTER, new SendCallback() {
            @Override
            public void onResult(int errorCode) {

            }
        });
        client.registerTagListener(getApplicationContext(), MessageTag.TAG_TEXT, listener);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        client.unregisterListener(getApplicationContext(), listener);
    }
}

File: app/src/main/java/com/example/androidlearn/MainActivity.kt
package com.example.androidlearn

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    companion object {
        const val TAG = "wenming"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.d(TAG, "MainActivity:onCreate")
    }

    fun jump(view: View) {
        startActivity(Intent(this, SecondActivity::class.java))
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "MainActivity:onStart")
    }


    override fun onResume() {
        super.onResume()
        Log.d(TAG, "MainActivity:onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "MainActivity:onPause")
    }


    override fun onStop() {
        super.onStop()
        Log.d(TAG, "MainActivity:onStop")
    }


    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "MainActivity:onDestroy")
    }


    override fun onNewIntent(p0: Intent?) {
        super.onNewIntent(p0)
        Log.d(TAG, "MainActivity:onNewIntent")
    }


}